pkg/collector/python/datadog_agent.go

```diff
@@ -609,3 +612,78 @@ func ObfuscateMongoDBString(cmd *C.char, errResult **C.char) *C.char {
 	// memory will be freed by caller
 	return TrackedCString(obfuscatedMongoDBString)
 }
+
+var (
+	telemetryHistograms = new(sync.Map)
+)
+
+func lazyInitTelemetryHistogram(checkName string, metricName string) telemetry.Histogram {
+	var key = checkName + "." + metricName
+	histogram, _ := telemetryHistograms.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewHistogramWithOpts(
+		checkName,
+		metricName,
+		nil,
+		fmt.Sprintf("Histogram of %s for Python check %s", metricName, checkName),
+		[]float64{10, 25, 50, 75, 100, 250, 500, 1000, 10000},
+		telemetry.DefaultOptions,
```

Comment:
Interesting it feels that map's value to store (if it is not there yet) would be always evaluated meaning created even when not used. But other approach to concurrent code is less elegant.

Reply:
... maybe two-liner load and if fails loa and store

Reply:
AFAIK `LoadOrStore` is the safe way to perform that operation on sync maps. The component and metric creation should be pretty fast and quickly discarded, especially at the relatively low rate these will be called.